packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -2067,15 +2067,24 @@ function updateContextConsumer(
 }
 
 function updateEventComponent(current, workInProgress, renderExpirationTime) {
-  const nextProps = workInProgress.pendingProps;
-  let nextChildren = nextProps.children;
+  const eventResponder = workInProgress.type.responder;
+  const getTransformedProps = eventResponder.getTransformedProps;
+  let nextPendingProps = workInProgress.pendingProps;
+  if (typeof getTransformedProps === 'function') {
+    workInProgress.pendingProps = nextPendingProps = getTransformedProps(
```

Comment:
I'm not sure this will actually remain safe. I'm about to add some code that does some minor resuming which might pick up these transformed props in a second pass over the component as the input.

Reply:
Okay, let me know what I need to change to make this work, assuming I can address your concerns with this API.